src/Connection/Cluster/RedisCluster.php

```diff
@@ -691,4 +705,51 @@ public function getParameters(): ?ParametersInterface
     {
         return $this->connectionParameters;
     }
+
+    /**
+     * Loop over sockets and reads incoming data.
+     *
+     * @return mixed
+     */
+    public function read()
+    {
+        $this->setSocketsNonBlockingMode();
+
+        $i = 0;
+        while (true) {
+            $socket = $this->sockets[$i]['socket'];
+            $socketArray = [$socket];
+            $write = null;
+            $except = null;
+            $num = stream_select($socketArray, $write, $except, 0);
```

Comment:
either use `stream_select` and pass it all the sockets in one call, or use `socket_select` if there is a reason to read them one by one

Reply:
It's some kind of event loop so it's constantly loop over streams if there's some data to read, so we can't process all of them in the same time. I know there's a confusion between streams and sockets here, the reason is that it's actually streams but in sockets context, that why it called sockets across the code, so I decided to keep this naming so it won't   confuse.

Reply:
@vladvildanov can't you do something like [this](https://www.php.net/manual/en/function.stream-select.php#122839) (run `stream_select` for all the open sockets, then if it returns a number > 0 loop on the sockets)

Reply:
@leibale I can, but still only 1 response per read request can be returned (in case if all sockets have data to read). PHP is sync script language, so only one response per object lifecycle

Reply:
@leibale So I can `stream_select` all of them, but in case if `$num > 0` it's still only one could be processed